workflow/saga.go

```diff
@@ -0,0 +1,62 @@
+package workflow
+
+type (
+    SagaOptions struct {
+        ParallelCompensation bool
+        ContinueWithError    bool
+    }
+
+    Saga interface {
```

Comment:
I propose an alternative interface that uses ctx to pass saga around:

sagaCtx := saga.newSaga(ctx, options)
...
saga.addCompensation(ctx, ....)
...
saga.Compensate(ctx)

This approach works with calls to downstream functions without need to pass Saga object explicitly as Context is passed everywhere already.